### PR TITLE
Add stat var (group) filtering based on data existence for given places

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: internal/server/statvar_group.go [hunk 1]
```diff
@@ -312,3 +343,36 @@ func computeSpecializedEntity(parentSvg string, childSvg string) string {
...
+	}
+	result := map[string]int{}
+	for _, rowKey := range rowList {
+		placeSv := keyTokens[rowKey]
+		token, _ := keyToTokenFn(rowKey)
+		if _, ok := baseDataMap[token]; ok {
+			result[placeSv.statVar]++
+		} else {
+			// TODO(shifucun): remove the else part once cache is build.
+			result[placeSv.statVar]++
```
Comment (shifucun) [inline internal/server/statvar_group.go:374]: This effectively making the filter an no-op. Will remove this after cache is build

File: internal/server/statvar_group.go [hunk 2]
```diff
@@ -187,18 +187,49 @@ func (s *Server) GetStatVarGroupNode(
...
+	if err != nil {
+		return nil, err
+	}
+	// Filter parent stat var groups
+	filteredParentStatVarGroups := []string{}
+	for _, item := range result.ParentStatVarGroups {
+		if c, ok := statExistence[item]; ok && c == len(places) {
+			filteredParentStatVarGroups = append(filteredParentStatVarGroups, item)
+		}
+	}
```
Comment (pradh) [inline internal/server/statvar_group.go:213]: Can this be a function perhaps? (maybe have TODO)

File: internal/server/statvar_search.go [hunk 3]
```diff
@@ -37,9 +39,31 @@ func (s *Server) SearchStatVar(
...
+			}
+		}
+		svList = filteredSvList
+		filteredSvgList := []string{}
+		for _, svg := range svgList {
+			if c, ok := statExistence[svg]; ok && c == len(places) {
+				filteredSvgList = append(filteredSvgList, svg)
+			}
+		}
+		svgList = filteredSvgList
```
Comment (pradh) [inline internal/server/statvar_search.go:63]: Can this be shared with the other check?

**Review by pradh — APPROVED** [review]

File: internal/server/statvar_group.go (hunk 2 above)
Comment (shifucun) [reply internal/server/statvar_group.go:214]: Added a to do for a generic way of filtering.

File: internal/server/statvar_search.go (hunk 3 above)
Comment (shifucun) [reply internal/server/statvar_search.go]: Done

**Comment by shifucun** [comment]
Thanks for review!

**Pull request merged**